chef/automate
# Adding project filters for GetRunsCounts

### :nut_and_bolt: Description

Adding project filtering for the cfgmgmt.proto GetRunsCounts RPC.

Limited the functionality of the GetRunsCounts RPC. Now an API user has to provide the node ID to use for the counts. A user can not get the run counts of more than one node. This does not change the behavior of the UI. 

### :athletic_shoe: Demo Script / Repro Steps
1. `build components/config-mgmt-service && build compontents/automate-gateway && build components/automate-ui && start_all_services`
1. `chef_load_nodes 10`
1. `chef_load_nodes 10`
1. Open `https://a2-dev.test/client-runs`
1. Select a node on the client runs page
1. Select the `Run History` on the left. 
1. Ensure that there are two runs. 
1. One can also check that in the Network inspect tab that all request with `run_counts` did not fail. 

### :chains: Related Resources

https://github.com/chef/automate/issues/68

### :white_check_mark: Checklist
